compiler/plugins/target/AMD-AIE/iree-amd-aie/Target/XCLBinGen.cpp

```diff
@@ -933,33 +989,26 @@ static LogicalResult generateUnifiedObject(
   if (useChess) {
     SmallString<64> inputLLChessHackedFile(tempDir);
     sys::path::append(inputLLChessHackedFile, "input.chesshacked.ll");
-    {
-      std::string inputLLStr;
-      auto inputLL = openInputFile(LLVMIRFile, &errorMessage);
-      if (!inputLL) {
-        llvm::errs() << "Failed to open input.ll file because: "
-                     << errorMessage;
-        return failure();
-      }
-      inputLLStr = std::string(inputLL->getBuffer());
-      std::string inputLLChessHackedStr = chesshack(inputLLStr);
-      if (auto maybeErr = dumpStrToDisk(inputLLChessHackedStr,
-                                        inputLLChessHackedFile.str().str());
-          maybeErr.has_value()) {
-        llvm::errs() << "Failed to dump to disk input.chesshacked.ll because: "
-                     << maybeErr;
-        return failure();
-      }
-    }
-
-    auto maybeVitisDir = findVitis(vitisDir);
-    if (failed(maybeVitisDir)) return failure();
-
-    if (failed(assembleUsingChess(inputLLChessHackedFile.c_str(), outputFile,
-                                  *vitisDir, tempDir, verbose))) {
-      llvm::errs() << "Failed to assemble input.o";
+    std::string inputLLStr;
+    auto inputLL = openInputFile(LLVMIRFile, &errorMessage);
+    if (!inputLL) {
+      llvm::errs() << "Failed to open input.ll file because: " << errorMessage;
       return failure();
     }
+    inputLLStr = std::string(inputLL->getBuffer());
+    std::string inputLLChessHackedStr = chesshack(inputLLStr);
+    FailureOr<std::string> maybeVitisDir = findVitis(vitisDir);
+    if (failed(maybeVitisDir)) return failure();
+    auto chessIntrinsicsObjFile = assembleStringUsingChess(
+        /*inputFileStr=*/inputLLChessHackedStr,
+        /*inputFileName=*/"input.chesshacked.ll",
+        /*outputFileName=*/outputFile,
+        /*outputDir=*/tempDir,
+        /*extraArgs*/ std::vector<std::string>{},
+        /*workDir=*/tempDir,
+        /*vitisDir=*/*maybeVitisDir,
+        /*verbose=*/verbose);
+    if (failed(chessIntrinsicsObjFile)) return failure();
   } else {
     SmallString<64> peanoOptBin(peanoDir);
```

Comment:
wondering if we can use `assembleStringUsingPeano` here?

Reply:
This is calling `llc` but I guess clang can just straight up compile/accept lol. Lemme try

Reply:
ah I see so chess uses same tool but we dont, from my past experience I dont think clang will be that nice to you (you can comnpile to llvm ir but not from that to obj in clang AFAIK), so feel free to just land as is.

Reply:
i couldn't manage to pass `--disable-builtin=memset` successfully (`-fno-builtin-memset` didn't work) but still a small reduction in boilerplate.